pkg/v1/tkg/client/machine_deployment_utkg.go

```diff
@@ -0,0 +1,284 @@
+// Copyright 2022 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package client
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+
+	v1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	capi "sigs.k8s.io/cluster-api/api/v1beta1"
+
+	"github.com/vmware-tanzu/tanzu-framework/pkg/v1/tkg/clusterclient"
+)
+
+func DoSetMachineDeploymentCC(clusterClient clusterclient.Client, cluster *capi.Cluster, options *SetMachineDeploymentOptions) error {
+	var update *capi.MachineDeploymentTopology
+	var base *capi.MachineDeploymentTopology
+
+	for i := range cluster.Spec.Topology.Workers.MachineDeployments {
```

Comment:
`cluster.Spec.Topology.Workers` may be `nil` and thus induce panic here.

Reply:
done